### ISSUE TITLE
how can I set cfg file for custom data set?

### ISSUE THREAD

**Issue by zpmmehrdad** [AlexeyAB/darknet#400]
hello...
I have 27 classes and there are 100 images per class.

[net]
batch=64
subdivisions=8
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate= ?
max_batches =  ?
steps=-?
scales=?

what should I set?

Thanks

**Comment by TheMikeyR** [comment]
You could start by copying the format of cfg/yolo-voc.2.0.cfg and then train with it and see how far you get, if you get unsatisfying results you could try to modify the network and see if it improves (trial and error)

**Comment by zpmmehrdad** [comment]
Thanks a lot.
I'm using yolo-voc.2.0.cfg now.and i set learning_rate up to 0.001 and max_batches = 5700
is it right?

**Comment by TheMikeyR** [comment]
It is recommended to train at least 2000*classes iterations, so in your case max_batches should be set to minimum 54000. 
I'm no expert in modifying learning_rate, but I would just try to go with default and see how the result goes, and then run it again with an higher learning rate.

**Comment by TheMikeyR** [comment]
I've edited my comment since I made a false claim.

**Comment by zpmmehrdad** [comment]
Thank you very much, I'll try it

**Comment by sivagnanamn** [comment]
@zpmmehrdad I usually set my maxbatch size with a thumb rule (2000 batches per class) i.e if I'm training for detecting 10 classes, I usually set 10x2000 = 20000 max batches with 64 images per batch. This setting seems to work fine for me so far. If your loss doesnt reduce, you can restart the training from the latest weights and train longer.

**Comment by zpmmehrdad** [comment]
@sivagnanamn it is much appreciated
